src/mpid/ch4/shm/xpmem/xpmem_types.h

```diff
@@ -41,23 +35,13 @@ typedef struct MPIDI_SHM_ctrl_xpmem_send_lmt_send_fin {
     uint64_t req_ptr;
 } MPIDI_SHM_ctrl_xpmem_send_lmt_send_fin_t;
 
+typedef struct MPIDI_SHM_ctrl_xpmem_send_lmt_recv_fin {
+    uint64_t req_ptr;
+} MPIDI_SHM_ctrl_xpmem_send_lmt_recv_fin_t;
+
 typedef struct MPIDI_SHM_ctrl_xpmem_send_lmt_cnt_free {
     int coop_counter_direct_flag;
     uint64_t coop_counter_offset;
 } MPIDI_SHM_ctrl_xpmem_send_lmt_cnt_free_t;
 
-typedef MPIDI_SHM_ctrl_xpmem_send_lmt_send_fin_t MPIDI_SHM_ctrl_xpmem_send_lmt_recv_fin_t;
-#endif
```

Comment:
I do not see the reason why we need change this line:
```typedef MPIDI_SHM_ctrl_xpmem_send_lmt_send_fin_t MPIDI_SHM_ctrl_xpmem_send_lmt_recv_fin_t;```

Reply:
Wearing the hat of active message, `send_fin_t` and `recv_fin_t` are two semantically unrelated packet. Aliasing them together not only forces you to use the non-specific member name `req_ptr` rather than `sreq_ptr` and `rreq_ptr` like everywhere else, it also makes two protocols less flexible. For example, if we need pass different amount of information to send_fin_t and recv_fin_t, the aliasing makes it confusing.

In another word, because they are semantically unrelated, I would like the code reflect that semantics.